instrumentation/bunny/lib/opentelemetry/instrumentation/bunny/patch_helpers.rb

```diff
@@ -0,0 +1,83 @@
+# frozen_string_literal: true
+
+# Copyright The OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+module OpenTelemetry
+  module Instrumentation
+    module Bunny
+      # The PatchHelper module provides functionality shared between patches.
+      #
+      # For additional details around trace messaging semantics
+      # See https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/trace/semantic_conventions/messaging.md#messaging-attributes
+      module PatchHelpers
+        def self.with_send_span(channel, tracer, exchange, routing_key, &block)
+          attributes = basic_attributes(channel, exchange, routing_key)
+          destination = destination_name(exchange, routing_key)
+
+          tracer.in_span("#{destination} send", attributes: attributes, kind: :producer, &block)
+        end
+
+        def self.with_receive_span(channel, tracer, delivery_info, properties, &block)
+          exchange = delivery_info.exchange
+          routing_key = delivery_info.routing_key
+          attributes = basic_attributes(channel, exchange, routing_key)
+          destination = destination_name(exchange, routing_key)
+          parent_context, links = extract_context(properties[:headers])
+
+          OpenTelemetry::Context.with_current(parent_context) do
+            tracer.in_span("#{destination} receive", links: links, attributes: attributes, kind: :consumer, &block)
+          end
+        end
+
+        def self.with_process_span(channel, tracer, delivery_info, properties, &block)
+          destination = destination_name(delivery_info[:exchange], delivery_info[:routing_key])
+          parent_context, links = extract_context(properties[:headers])
+
+          OpenTelemetry::Context.with_current(parent_context) do
+            tracer.in_span("#{destination} process", links: links, kind: :consumer, &block)
+          end
+        end
```

Comment:
There are some corner cases where `delivery_info` and `properties` can both be `nil` in both these methods. We should make sure we handle those safely.